app/controllers/bus_stops_controller.rb

```diff
@@ -77,6 +77,7 @@ def update
 
   def find_stop
     @stop = BusStop.find_by hastus_id: params.require(:id)
+    @data_fields = @stop.data_fields
```

Comment:
I wouldn't have bothered assigning this to its own instance variable personally.

Reply:
Actually reading further I see why you did it.

Reply:
Collect your data in the controller, was my thinking.